dataline-scheduler/src/main/java/io/dataline/scheduler/WorkerRunFactory.java

```diff
@@ -118,36 +97,33 @@ public void run() {
             // mediated in DefaultSyncWorker.
             new DefaultSyncWorker(
                 new SingerTapFactory(job.getConfig().getSync().getSourceDockerImage(), pbf, discoverSchemaWorker),
-                new SingerTargetFactory(job.getConfig().getSync().getDestinationDockerImage(), pbf)),
-            connectionPool)
-                .run();
-        break;
+                new SingerTargetFactory(job.getConfig().getSync().getDestinationDockerImage(), pbf)));
       default:
         throw new RuntimeException("Unexpected config type: " + job.getConfig().getConfigType());
     }
+
   }
 
   private static StandardCheckConnectionInput getCheckConnectionInput(JobCheckConnectionConfig config) {
-    final StandardCheckConnectionInput checkConnectionInput = new StandardCheckConnectionInput()
-        .withConnectionConfiguration(config.getConnectionConfiguration());
-
-    return checkConnectionInput;
+    return new StandardCheckConnectionInput().withConnectionConfiguration(config.getConnectionConfiguration());
   }
 
   private static StandardDiscoverSchemaInput getDiscoverSchemaInput(JobDiscoverSchemaConfig config) {
-    final StandardDiscoverSchemaInput discoverSchemaInput = new StandardDiscoverSchemaInput()
-        .withConnectionConfiguration(config.getConnectionConfiguration());
-
-    return discoverSchemaInput;
+    return new StandardDiscoverSchemaInput().withConnectionConfiguration(config.getConnectionConfiguration());
   }
 
   private static StandardSyncInput getSyncInput(JobSyncConfig config) {
-    final StandardSyncInput syncInput = new StandardSyncInput()
+    return new StandardSyncInput()
         .withSourceConnectionImplementation(config.getSourceConnectionImplementation())
         .withDestinationConnectionImplementation(config.getDestinationConnectionImplementation())
         .withStandardSync(config.getStandardSync());
+  }
+
+  @FunctionalInterface
+  interface Creator {
```

Comment:
is the main purpose of this interface to make testing easier? if so would it be okay to add a comment explaining why it exists? it feels a bit awkward but if it's mainly for testing then i'm cool with it, but would like to make that explicit.